openstack/clustering/v1/policies/requests.go

```diff
@@ -54,3 +60,73 @@ func List(client *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pa
 		return p
 	})
 }
+
+// CreateOpts params
+type CreateOpts struct {
+	Name string `json:"name"`
+	Spec Spec   `json:"spec"`
+}
+
+type Spec struct {
+	Description string                 `json:"description"`
+	Properties  map[string]interface{} `json:"properties"`
+	Type        string                 `json:"type"`
+	Version     Version                `json:"version"`
+}
+
+// ToPolicyCreateMap formats a CreateOpts into a body map.
+func (opts CreateOpts) ToPolicyCreateMap() (map[string]interface{}, error) {
+	b, err := gophercloud.BuildRequestBody(opts, "")
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]interface{}{"policy": b}, nil
+}
+
+type Version string
+
+// custom unmarshal function to handle Version returned as either float64 or string
+func (v *Version) UnmarshalJSON(b []byte) error {
+	if b[0] == '"' {
+		return json.Unmarshal(b, (*string)(v))
+	}
+
+	var f float64
+	if err := json.Unmarshal(b, &f); err != nil {
+		return err
+	}
+
+	if f == 1 {
+		version := fmt.Sprintf("%.1f", f)
+		*v = Version(version)
+		return nil
+	} else {
+		version := strconv.FormatFloat(f, 'f', -1, 64)
+		*v = Version(version)
+		return nil
+	}
+}
+
+// Create makes a request against the API to create a policy
+func Create(client *gophercloud.ServiceClient, opts CreateOpts) (r CreateResult) {
+	b, err := opts.ToPolicyCreateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	_, r.Err = client.Post(policyCreateURL(client), b, &r.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{201},
+	})
+	return
+}
+
+// Create makes a request against the API to delete a policy
+func Delete(client *gophercloud.ServiceClient, policyID string) (r PolicyDeleteResult) {
```

Comment:
https://github.com/gophercloud/gophercloud/blob/master/openstack/objectstorage/v1/containers/requests.go#L115-L119

Reply:
I'm trying it with the code you referenced, but the header is not populated.  It seems like you have to manually assign the r.Header like they do here:
https://github.com/gophercloud/gophercloud/blob/master/openstack/objectstorage/v1/containers/requests.go#L187

Reply:
That's interesting... I wonder if there's a bug in the containers Delete function then.